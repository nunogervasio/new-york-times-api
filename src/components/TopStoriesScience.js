import React, { Component } from "react";
import TopStorieCard from "./TopStoriesCard";
import { Header } from "semantic-ui-react";
import { sliceArticals } from "../utils";
import { APIKEY } from "../config";

class TopStoriesScience extends Component {
  state = {
    data: []
  };

  componentDidMount() {
    fetch(
      `https://api.nytimes.com/svc/topstories/v2/science.json?api-key=${APIKEY}`
    )
      .then(responce => responce.json())
      .then(responceData =>
        this.setState({
          data: responceData.results
        })
      )
      .catch(error => {
        console.log("Error fetching and parsing data...", error);
      });
  }
  render() {
    return (
      <div>
        <div className="section-header">
          <Header size="large" textAlign="center">
            Science
          </Header>
        </div>
        {sliceArticals(this.state.data, 3).map((obj, index) => (
          <TopStorieCard
            key={index}
            artical_url={obj.url}
            headline={obj.title}
            summary={obj.abstract}
            pub_date={obj.published_date}
            // img_url={obj.multimedia[3].url}
            divider_icon="hand spock"
          />
        ))}
      </div>
    );
  }
}

export default TopStoriesScience;
