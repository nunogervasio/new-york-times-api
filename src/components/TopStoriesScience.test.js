import React from "react";
import { shallow } from "enzyme";
import TopStoriesScience from "./TopStoriesScience";

describe("TopStoriesScience", () => {
  const topStoriesScience = shallow(<TopStoriesScience />);

  it("renders properly", () => {
    expect(topStoriesScience).toMatchSnapshot();
  });
  it("renders TopStoriesCard", () => {});
});
