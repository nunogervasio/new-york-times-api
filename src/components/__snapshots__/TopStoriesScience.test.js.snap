// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopStoriesScience renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopStoriesScience />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="section-header"
        >
          <Header
            size="large"
            textAlign="center"
          >
            Science
          </Header>
        </div>,
        Array [],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Header
            size="large"
            textAlign="center"
          >
            Science
          </Header>,
          "className": "section-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Science",
            "size": "large",
            "textAlign": "center",
          },
          "ref": null,
          "rendered": "Science",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="section-header"
          >
            <Header
              size="large"
              textAlign="center"
            >
              Science
            </Header>
          </div>,
          Array [],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Header
              size="large"
              textAlign="center"
            >
              Science
            </Header>,
            "className": "section-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Science",
              "size": "large",
              "textAlign": "center",
            },
            "ref": null,
            "rendered": "Science",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
